app/src/main/java/com/fcaputo/parkingapp/mvp/presenter/SettingsPresenter.kt

```diff
@@ -0,0 +1,33 @@
+package com.fcaputo.parkingapp.mvp.presenter
+
+import com.fcaputo.parkingapp.mvp.contract.SettingsContract
+import com.fcaputo.parkingapp.utils.Constants
+import com.fcaputo.parkingapp.utils.validation.ValidationErrorType
+
+class SettingsPresenter(private val model: SettingsContract.Model, private val view: SettingsContract.View) : SettingsContract.Presenter {
+    init {
+        view.onSaveButton { onSaveButton() }
+        view.showParkingSize(getSizeString(model.getParkingSize()))
+    }
+
+    override fun onSaveButton() {
+        if(formIsIncomplete()) {
+            view.showErrorMessage(ValidationErrorType.INCOMPLETE_FIELD)
+            return
+        }
+
+        val size = view.getParkingSize()
+        val result = model.validate(size)
+        if (result.isSuccess){
+            model.save(size)
+            view.showSuccessMessage()
+            view.showParkingSize(getSizeString(model.getParkingSize()))
+            view.clear()
+        } else {
+            result.error?.let { view.showErrorMessage(it) }
+        }
+    }
+
+    private fun formIsIncomplete(): Boolean = view.getTextFieldsContents().any{ it.isNullOrEmpty() }
+    private fun getSizeString(size: Int): String = if (size != Constants.SIZE_NOT_SET_INT) size.toString() else "-"
```

Comment:
Do not hardcode, use constants or string res if correspond

Reply:
Ok, got it

Reply:
Done